tokendito/helpers.py

```diff
@@ -442,14 +454,33 @@ def user_configuration_input():
 
     """
     logging.debug("Obtain user input for the user.")
-
-    all_config_msgs = ['Okta App URL. E.g https://acme.okta.com/home/'
-                       'amazon_aws/b07384d113edec49eaa6/123: ',
-                       'Organization username. E.g jane.doe@acme.com: ']
+    url = ''
+    username = ''
     config_details = []
-    for config_msg in all_config_msgs:
-        user_input = to_unicode(input(config_msg))
-        config_details.append(user_input)
+    message = {
+        'app_url': '\nOkta App URL. E.g https://acme.okta.com/home/'
+                   'amazon_aws/b07384d113edec49eaa6/123\n[none]: ',
+        'username': '\nOrganization username. E.g jane.doe@acme.com'
+                    '\n[none]: '
+    }
+
+    while url == '':
+        user_data = to_unicode(input(message['app_url']))
+        user_data = user_data.strip()
+        if validate_okta_aws_app_url(input_url=user_data):
```

Comment:
validate_okta_aws_app_url(user_data)

Reply:
We are checking whether or not the URL is valid. `validate_okta_aws_app_url` returns a boolean, so we need to see whether the validation passed.

Reply:
I just think (input_url=user_data) is superfluous, 
 if validate_okta_aws_app_url(user_data):
To make things even clearer, maybe change the var name from user_data to user_data_app_url or input_app_url.

Reply:
ah! understood. removed the assignment during the function call.